tests/Service/CollapsedBroadcastsService/FilterCategoriesByBroadcastedDateTest.php

```diff
@@ -1,40 +1,71 @@
+    public function testRepositoryReceiveNeededParamsToFetchBroadcastsByDate()
     {
-        $dbAncestry1 = [1, 2, 3];
-        $category1 = $this->mockEntity('Genre', 3);
-        $category1->method('getDbAncestryIds')->willReturn($dbAncestry1);
+        $stubCat1 = $this->createConfiguredMock(Genre::class, ['getDbAncestryIds' => [1, 2, 3]]);
+        $stubCat2 = $this->createConfiguredMock(Genre::class, ['getDbAncestryIds' => [5, 6, 7]]);
 
-        $dbAncestry2 = [5, 6, 7];
-        $category2 = $this->mockEntity('Genre', 5);
-        $category2->method('getDbAncestryIds')->willReturn($dbAncestry2);
+        $startDateTime = new DateTimeImmutable();
+        $endDateTime = new DateTimeImmutable();
 
-        $mockBroadcastedResults = [
-            ['ancestry' => '1,2,3,'],
-            ['ancestry' => '1,2,3,4,'],
-        ];
+        $this->mockRepository
+            ->expects($this->once())
+            ->method('filterCategoriesByBroadcastedDates')
+            ->with(
+                [$stubCat1->getDbAncestryIds(), $stubCat2->getDbAncestryIds()],
+                false,
+                $startDateTime,
+                $endDateTime
+            );
 
-        $start = new DateTimeImmutable();
-        $end = new DateTimeImmutable();
+        $this->service()->filterCategoriesByBroadcastedDate([$stubCat1, $stubCat2], $startDateTime, $endDateTime);
+    }
 
-        $this->mockRepository->expects($this->once())
-            ->method('filterCategoriesByBroadcastedDates')
-            ->with([$dbAncestry1, $dbAncestry2], false, $start, $end)
-            ->willReturn($mockBroadcastedResults);
+    /**
+     * @dataProvider categoriesBroadcastedOnDateProvider
+     */
+    public function testCategoriesBroadcastedOnDateAreFiltered($expectedCategoriesFoundOnDate, $dbResults)
+    {
+        $this->mockRepository->method('filterCategoriesByBroadcastedDates')->willReturn($dbResults);
 
-        $dbBroadcastedCategories = $this->service()->filterCategoriesByBroadcastedDate(
-            [$category1, $category2],
-            $start,
-            $end
+        $categories = $this->service()->filterCategoriesByBroadcastedDate(
+            [
+                $this->createConfiguredMock(Genre::class, ['getDbAncestryIds' => [1, 2, 3]]),
+                $this->createConfiguredMock(Genre::class, ['getDbAncestryIds' => [19, 20, 30, 40]]),
+            ],
+            new DateTimeImmutable(),
+            new DateTimeImmutable()
         );
 
-        $this->assertCount(1, $dbBroadcastedCategories);
-        $this->assertSame([1, 2, 3], $dbBroadcastedCategories[0]->getDbAncestryIds());
+        $this->assertContainsOnly(Genre::class, $categories);
+        $this->assertEquals($expectedCategoriesFoundOnDate, $this->extractDbAncestryIds($categories));
+    }
+
+    public function categoriesBroadcastedOnDateProvider(): array
+    {
+        return [
+            'CASE: none of category specified was not broadcasted on date' => [
```

Comment:
This seems to have the same key as the test case below

Also double negatives sound weird, is there a better way to phrase this?

Reply:
Yeah!, phpstan was handy here, I just fixed it